src/pages/home/sidebar/SidebarNavigationContext.js

```diff
@@ -0,0 +1,58 @@
+import React, {useMemo, useCallback, useState} from 'react';
+import PropTypes from 'prop-types';
+import CONST from '../../../CONST';
+import Navigation from '../../../libs/Navigation/Navigation';
+import GLOBAL_NAVIGATION_MAPPING from '../../../GLOBAL_NAVIGATION_MAPPING';
+
+const propTypes = {
+    children: PropTypes.node.isRequired,
+};
+
+const SidebarNavigationContext = React.createContext({
+    selectedGlobalNavigationOption: undefined,
+    selectedSubNavigationOption: undefined,
+    updateFromNavigationState: () => {},
+});
```

Comment:
The key here is onStateChange. Not only do we want a state, but we also need to know when it changed

Reply:
Yes, you're right 😄  I completely forgot about this hook.

Let's do it this way. We'll keep the context for now. It may be useful for the implementation of the second part if something unpredictable comes out. If not, I'll change it to use the useNavigationState hook before merging the second part. Ok?

Reply:
that sounds like a plan! thanks for the update